cmd/kube-apiserver/app/server.go

```diff
@@ -563,10 +571,10 @@ func buildGenericConfig(
 }
 
 // BuildAuthenticator constructs the authenticator
-func BuildAuthenticator(s *options.ServerRunOptions, EgressSelector *egressselector.EgressSelector, extclient clientgoclientset.Interface, versionedInformer clientgoinformers.SharedInformerFactory) (authenticator.Request, *spec.SecurityDefinitions, error) {
+func BuildAuthenticator(s *options.ServerRunOptions, EgressSelector *egressselector.EgressSelector, extclient clientgoclientset.Interface, versionedInformer clientgoinformers.SharedInformerFactory) (authenticator.Request, *spec.SecurityDefinitions, func(*genericapiserver.Config) error, error) {
```

Comment:
from the callsite, you  have the GenericConfig.  Why not simply pass it through to this method and directly add inline?

Reply:
I suppose that statement would apply even to the preexisting code.